kore/src/Kore/Log.hs

```diff
@@ -284,30 +276,19 @@ makeKoreLogger exeName timestampSwitch logToText =
             typeName = type' someEntry'
     indent = Pretty.indent 4
 
+getElapsedTime :: MonadIO io => io RtsTime
+getElapsedTime =
+    liftIO $ fmap (toMicroseconds . Stats.elapsed_ns) Stats.getStats
```

Comment:
Are you sure this gets the correct time? It seems like [elapsed_ns](https://hackage.haskell.org/package/base-4.14.0.0/docs/GHC-Stats.html#v:elapsed_ns) is specific to garbage collection. Also, are we sure we want to use this package? This is in the description of the module:
```
This module provides access to internal garbage collection and memory usage statistics. These statistics are not available unless a program is run with the -T RTS flag.

This module is GHC-only and should not be considered portable.
```

Reply:
I took another look after you told me you used the functions from our `Stats` module, but I'm still unsure of how safe this is. It looks like `getStats` explicitly triggers a GC, and this side effect might impact execution in an unintended way. @ttuegel , when you have time, could you take a quick look as well?

Reply:
That's right, you can't use `Stats` here. Look at how [`clockSomethingIO`](https://github.com/kframework/kore/blob/ce2b5beeab5cd7f7373a10ef9b2b62fd28ea6123/kore/app/share/GlobalMain.hs#L422) is implemented. The start time needs to be acquired before the logger is created.